### PR TITLE
porting statistics script

### PR DESCRIPTION
**Pull request by anishganti**

### CODE REVIEW

File: src/juicebox/tools/clt/CLTFactory.java [hunk 1]
```diff
@@ -135,8 +135,11 @@ public static JuiceboxCLT getCLTCommand(String cmd) {
             return new LibraryComplexity();
         } else if (cmd.equals("apa_vs_distance")) { //Todo check if okay
             return new APAvsDistance();
+        } else if (cmd.equals("statisticsscript")) {
```
Comment (sa501428) [inline src/juicebox/tools/clt/CLTFactory.java]: let's change the name to statistics

File: src/juicebox/tools/clt/old/StatisticsScript.java [hunk 2]
```diff
@@ -0,0 +1,593 @@
...
+import juicebox.tools.clt.JuiceboxCLT;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.text.NumberFormat;
+import java.util.*;
+
+public class StatisticsScript extends JuiceboxCLT {
+    //variables for getting parameters of input file, flags set to default initially
+    String outFile;
```
Comment (sa501428) [inline src/juicebox/tools/clt/old/StatisticsScript.java]: set permission level for all these parameters; they should probably all be private. some should also be final (the ones that are defined only once and should not change)

File: src/juicebox/tools/clt/CLTFactory.java (hunk 1 above)
Comment (anishganti) [reply src/juicebox/tools/clt/CLTFactory.java:138]: will change it right now

File: src/juicebox/tools/clt/old/Statistics.java [hunk 3]
```diff
@@ -0,0 +1,590 @@
...
+    private static final int mapqValThreshold = 200;
+    private static final long[] bins = {10,12,15,19,23,28,35,43,53,66,81,100,123,152,187,231,285,351,433,534,658,811,1000,1233,1520,1874,2310,2848,3511,4329,5337,6579,8111,10000,12328,15199,18738,23101,28480,35112,43288,53367,65793,81113,100000,123285,151991,187382,231013,284804,351119,432876,533670,657933,811131,1000000,1232847,1519911,1873817,2310130,2848036,3511192,4328761,5336699,6579332,8111308,10000000,12328467,15199111,18738174,23101297,28480359,35111917,43287613,53366992,65793322,81113083,100000000,123284674,151991108,187381742,231012970,284803587,351119173,432876128,533669923,657933225,811130831,1000000000,1232846739,1519911083,1873817423,2310129700L,2848035868L,3511191734L,4328761281L,5336699231L,6579332247L,8111308308L,10000000000L};
+
+    public Statistics(){
+        //constructor
+        super(getUsage());
+    }
+
+    public static String getUsage(){
+        return  " Usage: StatisticsScript [--ligation NNNN] [-q mapq] <site file> <stats file> <infile> [outfile]\n" +
```
Comment (sa501428) [inline src/juicebox/tools/clt/old/Statistics.java:90]: this should also just say statistics for the common name (an in lower case)

File: src/juicebox/tools/clt/old/Statistics.java [hunk 4]
```diff
@@ -0,0 +1,590 @@
...
+                " <site file>: list of HindIII restriction sites, one line per chromosome\n" +
+                " <stats file>: output file containing total reads, for library complexity\n"+
+                " <infile>: file in intermediate format to calculate statistics on, can be stream\n" +
+                " [outfile]: output, results of fragment search\n";
+    }
+
+    public void readSiteFile(){
+        //read in restriction site file and store as multidimensional array q
+        try{
+            BufferedReader restriction_site = new BufferedReader(new FileReader(siteFile));
```
Comment (sa501428) [inline src/juicebox/tools/clt/old/Statistics.java:102]: camelCase

File: src/juicebox/tools/clt/old/Statistics.java [hunk 5]
```diff
@@ -0,0 +1,590 @@
...
+    public void infileStatistics(){
+        //read in infile and calculate statistics
+        String danglingJunction = ligationJunction.substring(ligationJunction.length()/2);
+        try {
+            BufferedReader files = new BufferedReader(new FileReader(inFile));
+            String file = files.readLine();
+            while (file != null) {
+                unique++;
+                String[] record = file.split("\\s+");
+                int num_records = record.length;
```
Comment (sa501428) [inline src/juicebox/tools/clt/old/Statistics.java:136]: do a quick search for '_' and switch everything to camelcase

**Pull request closed without merging**